Lib9c/Action/SellCancellation.cs

```diff
@@ -132,8 +132,21 @@ public override IAccountStateDelta Execute(IActionContext context)
             }
 
             Order order = OrderFactory.Deserialize(orderDict);
-            order.ValidateCancelOrder(avatarState, tradableId);
-            ITradableItem sellItem = order.Cancel(avatarState, context.BlockIndex);
+            bool backward = false;
+            try
+            {
+                order.ValidateCancelOrder2(avatarState, tradableId);
+            }
+            catch (Exception)
+            {
+                //backward for not expired prev order.
+                order.ValidateCancelOrder(avatarState, tradableId);
+                backward = true;
+            }
+
+            var sellItem = backward
+                ? order.Cancel(avatarState, context.BlockIndex)
+                : order.Cancel2(avatarState, context.BlockIndex);
```

Comment:
IMO, `backward` is too ambiguous.

```suggestion
            bool fromPreviousAction = false;
            try
            {
                order.ValidateCancelOrder2(avatarState, tradableId);
            }
            catch (Exception)
            {
                //backward for not expired prev order.
                order.ValidateCancelOrder(avatarState, tradableId);
                fromPreviousAction = true;
            }

            var sellItem = fromPreviousAction
                ? order.Cancel(avatarState, context.BlockIndex)
                : order.Cancel2(avatarState, context.BlockIndex);
```